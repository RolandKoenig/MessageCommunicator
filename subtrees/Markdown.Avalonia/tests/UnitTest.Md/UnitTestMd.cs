using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using UnitTest.Base;
using UnitTest.Base.Utils;

namespace UnitTest.Md
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTestMd : UnitTestBase
    {
        [Test]
        [RunOnUI]
        public void Transform_givenTest1_generatesExpectedResult()
        {
            var text = Util.LoadText("Test1.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLists1_generatesExpectedResult()
        {
            var text = Util.LoadText("Lists1.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLists2_generatesExpectedResult()
        {
            var text = Util.LoadText("Lists2.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLists3_generatesExpectedResult()
        {
            var text = Util.LoadText("Lists3.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenTables1_generatesExpectedResult()
        {
            var text = Util.LoadText("Tables.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenHorizontalRules_generatesExpectedResult()
        {
            var text = Util.LoadText("HorizontalRules.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLinksInline_generatesExpectedResult()
        {
            var text = Util.LoadText("Links_inline_style.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenTextStyles_generatesExpectedResult()
        {
            var text = Util.LoadText("Text_style.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenImages_generatesExpectedResult()
        {
            var text = Util.LoadText("Images.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenBlockqoute_generatesExpectedResult()
        {
            var text = Util.LoadText("Blockquite.md");
            var markdown = new Markdown.Avalonia.Markdown();

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenMixing_generatesExpectedResult()
        {
            var text = Util.LoadText("Mixing.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenCodes_generatesExpectedResult()
        {
            var text = Util.LoadText("Codes.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenEmoji()
        {
            var text = Util.LoadText("Emoji.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }
    }

}