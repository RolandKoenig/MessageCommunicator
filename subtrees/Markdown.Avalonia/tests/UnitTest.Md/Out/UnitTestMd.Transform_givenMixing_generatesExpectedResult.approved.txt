<?xml version="1.0" encoding="utf-16"?>
<StackPanel xmlns:c="clr-namespace:ColorTextBlock.Avalonia;assembly=ColorTextBlock.Avalonia" xmlns:m="clr-namespace:Markdown.Avalonia.Controls;assembly=Markdown.Avalonia" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns="https://github.com/avaloniaui">
  <c:CTextBlock Classes="Heading1">
    <c:CRun Text="Markdown.Xaml" />
  </c:CTextBlock>
  <c:CTextBlock Classes="Paragraph">
    <c:CRun Text="Markdown XAML is a port of the popular " />
    <c:CItalic>
      <c:CRun Text="MarkdownSharp" />
    </c:CItalic>
    <c:CRun Text=" Markdown processor, but with one very significant difference: Instead of rendering to a string containing HTML, it renders to a FlowDocument suitable for embedding into a WPF window or usercontrol." />
  </c:CTextBlock>
  <c:CTextBlock Classes="Heading2">
    <c:CRun Text="Features" />
  </c:CTextBlock>
  <c:CTextBlock Classes="Paragraph">
    <c:CRun Text="MarkDown.Xaml has a number of convenient features" />
  </c:CTextBlock>
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="The engine itself is a single file, for easy inclusion in your own projects" />
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="1">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="1">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Code for the engine is structured in the same manner as the original MarkdownSharp" />
        <c:CLineBreak Text="&#xA;" />
        <c:CRun Text="If there are any bug fixes to the regular expressions in MarkdownSharp, merging those fixes in the Markdown.Xaml should be straightforward" />
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="2">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="2">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Includes a " />
        <c:CCode>
          <c:CRun Text="TextToFlowDocumentConverter" />
        </c:CCode>
        <c:CRun Text=" to make it easy to bind Markdown text" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
      <RowDefinition />
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <c:CTextBlock Classes="Heading2">
    <c:CRun Text="Markdown capabilities and customizables styles" />
  </c:CTextBlock>
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Links " />
        <c:CHyperlink HoverBackground="{x:Null}" HoverForeground="{x:Null}" CommandParameter="https://www.google.com">
          <c:CRun Text="Go to Google!" />
        </c:CHyperlink>
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="1">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="1">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Links with title " />
        <c:CHyperlink HoverBackground="{x:Null}" HoverForeground="{x:Null}" CommandParameter="https://www.google.com">
          <c:CRun Text="Go to Google!" />
        </c:CHyperlink>
      </c:CTextBlock>
    </StackPanel>
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="2">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="2">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Remote images" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
      <RowDefinition />
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <c:CTextBlock Classes="Paragraph">
    <c:CImage />
  </c:CTextBlock>
  <c:CTextBlock Classes="Paragraph">
    <c:CImage Classes="blue" />
    <c:CImage Classes="cyan" />
  </c:CTextBlock>
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Local images" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <c:CTextBlock Classes="Paragraph">
    <c:CImage />
    <c:CRun Text=" " />
    <c:CImage />
  </c:CTextBlock>
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Table" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <c:CTextBlock Classes="Paragraph">
    <c:CRun Text="table begin string" />
  </c:CTextBlock>
  <Border Classes="Table" xmlns="">
    <Border.Child>
      <Grid Classes="Table">
        <Border Classes="TableHeader" Grid.Column="0" Grid.Row="0">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="a" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="TableHeader" Grid.Column="1" Grid.Row="0">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="b" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="TableHeader" Grid.Column="2" Grid.Row="0">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="c" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="TableHeader" Grid.Column="3" Grid.Row="0">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="d" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="a1234567890" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="b1234567890" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="c1234567890" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="3" Grid.Row="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="d1234567890" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="0" Grid.Row="2">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="a" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="1" Grid.Row="2">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="b" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="2" Grid.Row="2">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="c" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="2">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="d" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="3">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="A" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="3">
          <Border.Child>
            <c:CTextBlock />
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="3">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="C" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="3" Grid.Row="3" />
        <Border Classes="EvenTableRow" Grid.Column="0" Grid.Row="4">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="1" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="1" Grid.Row="4">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="2" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="2" Grid.Row="4">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="3" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="4">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="4" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="5">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="あ" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="5">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="い" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="5">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="う" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="3" Grid.Row="5">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="え" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Grid.ColumnDefinitions>
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
        </Grid.RowDefinitions>
      </Grid>
    </Border.Child>
  </Border>
  <c:CTextBlock Classes="Paragraph">
    <c:CRun Text="table end string" />
  </c:CTextBlock>
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Code" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <c:CTextBlock Classes="Paragraph">
    <c:CRun Text="Markdown.Xaml support " />
    <c:CCode>
      <c:CRun Text="inline code" />
    </c:CCode>
  </c:CTextBlock>
  <Border Classes="CodeBlock" xmlns="">
    <Border.Child>
      <ScrollViewer Classes="CodeBlock">
        <ScrollViewer.Content>
          <TextBlock Text="#include &lt;stdio.h&gt;&#xA;int main()&#xA;{&#xA;   // printf() displays the string inside quotation&#xA;   printf(&quot;Hello, World!&quot;);&#xA;   return 0;&#xA;}" Classes="CodeBlock" />
        </ScrollViewer.Content>
      </ScrollViewer>
    </Border.Child>
  </Border>
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Separator" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <m:Rule BoldLineWidth="2" Classes="Bold" />
  <Grid Classes="List" xmlns="">
    <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
      <c:CRun Text="•" />
    </c:CTextBlock>
    <StackPanel Grid.Column="1" Grid.Row="0">
      <c:CTextBlock Classes="Paragraph">
        <c:CRun Text="Blockquote" />
      </c:CTextBlock>
    </StackPanel>
    <Grid.ColumnDefinitions>
      <ColumnDefinition />
      <ColumnDefinition />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
      <RowDefinition />
    </Grid.RowDefinitions>
  </Grid>
  <Border Classes="Blockquote" xmlns="">
    <Border.Child>
      <StackPanel Classes="Blockquote">
        <c:CTextBlock Classes="Heading2">
          <c:CRun Text="Features" />
        </c:CTextBlock>
        <c:CTextBlock Classes="Paragraph">
          <c:CRun Text="MarkDown.Xaml has a number of convenient features" />
        </c:CTextBlock>
        <Grid Classes="List">
          <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
            <c:CRun Text="•" />
          </c:CTextBlock>
          <StackPanel Grid.Column="1" Grid.Row="0">
            <c:CTextBlock Classes="Paragraph">
              <c:CRun Text="The engine itself is a single file, for easy inclusion in your own projects" />
            </c:CTextBlock>
          </StackPanel>
          <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="1">
            <c:CRun Text="•" />
          </c:CTextBlock>
          <StackPanel Grid.Column="1" Grid.Row="1">
            <c:CTextBlock Classes="Paragraph">
              <c:CRun Text="Code for the engine is structured in the same manner as the original MarkdownSharp " />
            </c:CTextBlock>
          </StackPanel>
          <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="2">
            <c:CRun Text="•" />
          </c:CTextBlock>
          <StackPanel Grid.Column="1" Grid.Row="2">
            <c:CTextBlock Classes="Paragraph">
              <c:CRun Text="Includes a " />
              <c:CCode>
                <c:CRun Text="TextToFlowDocumentConverter" />
              </c:CCode>
              <c:CRun Text=" to make it easy to bind Markdown text" />
            </c:CTextBlock>
          </StackPanel>
          <Grid.ColumnDefinitions>
            <ColumnDefinition />
            <ColumnDefinition />
          </Grid.ColumnDefinitions>
          <Grid.RowDefinitions>
            <RowDefinition />
            <RowDefinition />
            <RowDefinition />
          </Grid.RowDefinitions>
        </Grid>
      </StackPanel>
    </Border.Child>
  </Border>
  <c:CTextBlock Classes="Heading2">
    <c:CRun Text="What is this Demo?" />
  </c:CTextBlock>
  <c:CTextBlock Classes="Paragraph">
    <c:CRun Text="This demo application shows MarkDown.Xaml in use - as you make changes to the left pane, the rendered MarkDown will appear in the right pane." />
  </c:CTextBlock>
  <c:CTextBlock Classes="Heading3">
    <c:CRun Text="Source" />
  </c:CTextBlock>
  <c:CTextBlock Classes="Paragraph">
    <c:CRun Text="Review the source for this demo to see how MarkDown.Xaml works in practice, how to use the TextToFlowDocumentConverter, and how to style the output to appear the way you desire." />
  </c:CTextBlock>
</StackPanel>
